algviz/parser/structures.py

```diff
@@ -213,11 +214,33 @@ class TreeNode(DataStructure):
     """A node with some number of children in a fixed order.  Edges are implicit."""
     # A common superclass could be used for linked-list nodes, since linked
     # lists are just skinny trees
+
     def __init__(self, data, children=None, **kwargs):
         super().__init__(**kwargs)
         self.data = data
         self.children = [] if children is None else children
 
+    def is_leaf(self):
+        return len(self.children) == 0
+        
+    def tree_width(self):
+        if self.is_leaf():
+            return 1
+        width = max(len(self.children),1)
+        for child in self.children:
+            child_width = child.tree_width()
+            width = max(width, child_width)
+        return width
+
+    def tree_height(self):
```

Comment:
Either have it be a method named `height` or define it somewhere else as `tree_height`.  The `tree` part is obvious.